pallet/nft/src/impls.rs

```diff
@@ -376,6 +377,58 @@ impl<T: Config> Pallet<T> {
 		(new_cursor, total_owned, response)
 	}
 
+	/// Find the tokens details for the given collection id
+	/// Returns collection owner, name, metadata schema, max issuance,
+	/// next available serial number, collection issuance, is_cross_chain_compatible
+	pub fn collection_details(collection_id: CollectionUuid) -> CollectionDetail<T::AccountId>
+	where
+		<T as frame_system::Config>::AccountId: core::default::Default,
+	{
+		let collection_info = <CollectionInfo<T>>::get(collection_id);
+		if collection_info.is_none() {
+			// should not happen
+			log!(warn, "🃏 Unexpected empty collection: {:?}", collection_id);
+			let collection_info = CollectionDetail {
+				owner: T::AccountId::from(Default::default()),
+				royalties_schedule: Default::default(),
+				max_issuance: None,
+				origin_chain: Default::default(),
+				next_serial_number: 0,
+				collection_issuance: 0,
+				cross_chain_compatibility: Default::default(),
+				name: Default::default(),
+				metadata_scheme: Default::default(), // MetadataScheme::try_from(b"".as_slice()).unwrap()
+			};
+			return collection_info;
+		}
+		let collection_info = collection_info.unwrap();
+		let owner = collection_info.owner;
+		let name = collection_info.name.into();
+		let metadata_scheme = collection_info.metadata_scheme.construct_token_uri(0);
```

Comment:
nit, maybe instead of appending 0 we could return the raw token uri? 
```suggestion
		let metadata_scheme = collection_info.metadata_scheme.0.into_inner();
```